src/test/java/org/jenkinsci/plugins/runselector/selectors/ParameterizedRunSelectorTest.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * The MIT License
+ * 
+ * Copyright (c) 2015 IKEDA Yasuyuki
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.runselector.selectors;
+
+import hudson.model.Cause;
+import hudson.model.FreeStyleProject;
+import hudson.model.ParameterDefinition;
+import hudson.model.ParametersAction;
+import hudson.model.ParametersDefinitionProperty;
+import hudson.model.Run;
+import hudson.model.StringParameterDefinition;
+import hudson.model.StringParameterValue;
+import hudson.model.TaskListener;
+import org.jenkinsci.plugins.runselector.RunSelector;
+import org.jenkinsci.plugins.runselector.context.RunSelectorContext;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for {@link ParameterizedRunSelector}
+ */
+public class ParameterizedRunSelectorTest {
+
+    @ClassRule
+    public static JenkinsRule j = new JenkinsRule();
+
+    /**
+     * Should not cause a fatal error even for an undefined variable.
+     */
+    @Issue("JENKINS-30357")
+    @Test
+    public void testUndefinedParameter() throws Exception {
+        FreeStyleProject jobToSelect = j.createFreeStyleProject();
+        FreeStyleProject selecter = j.createFreeStyleProject();
+
+        RunSelector selector = new ParameterizedRunSelector("NoSuchVariable");
```

Comment:
I think this test is no longer necessary as `ParameterizedRunSelector` now only supports variable expressions and immediate values and no longer supports variable names.